### PR TITLE
Added video doorbell support

### PR DESCRIPTION
**Pull request by AlexanderBabel**
Hey there,

I implemented the video doorbell service. Feel free to try it. In my case it works very well.

Greetings
Alex

### CODE REVIEW

File: ffmpeg.js [hunk 1]
```diff
@@ -0,0 +1,381 @@
+'use strict';
```
Comment (AlexanderBabel) [inline ffmpeg.js:1]: This file is from this repo: https://github.com/KhaosT/homebridge-camera-ffmpeg

**Comment by krmkrx** [comment]
I got a problem while installing the camera accessory. I inserted the same code as the one of homebridge but however even after 3 minutes, the spinning loading wheel while adding the accessory was still there. I aborted the installation of the camera but I am unable to give it another try installing, since it does not appear anymore under the devices to install section. 

I have also tried the following: uninstalled your fork, removed nello from config.json, removed cachedAccessoires folder. No success. 

Any idea on how to be able to install the camera again after aborted installation?

The backend seems to be just working fine as per the log output:

```
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] [nello.io] Signed in.
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] [nello.io] Getting locations from nello.io.
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] [nello.io] Adding new accessory with location ID xxx.
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] [nello.io] Accessory name for location ID xxx is xxx.
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] [nello.io] Configuring accessory with location ID xxx.
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] [nello.io] Accessory for location with ID xxx added.
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] [nello.io] Got locations from nello.io.
Jun 09 18:26:57 raspberrypi homebridge[13348]: [6/9/2018, 6:26:57 PM] xxx Camera is running on port 43705.
Jun 09 18:26:58 raspberrypi homebridge[13348]: [6/9/2018, 6:26:58 PM] [nello.io] Updating webhook for door with ID xxx to https://cuddly-horse-40.localtunnel.me.
Jun 09 18:26:58 raspberrypi homebridge[13348]: [6/9/2018, 6:26:58 PM] [nello.io] Updated webhook for door with ID xxx to https://cuddly-horse-40.localtunnel.me.
```

When I ring the bell, it shows in the log:

`Jun 09 18:28:38 raspberrypi homebridge[13348]: [6/9/2018, 6:28:38 PM] [nello.io] Someone rang the bell of the door with ID xxx.`

**Comment by AlexanderBabel** [comment]
Just try a complete rest of your Homebridge instance. Including unpairing in the home app.

**Comment by krmkrx** [comment]
Thank you. However the notification does not contain a button to unlock the door.
Both values for doorbell and videoDoorbell are set to true.

Could you explain the difference between doorbell and videoDoorbell more detailed?
As far as I understood, setting doorbell to true only provides a notification whereas setting videoDoorbell to true should provide the button to unlock, too.

**Comment by AlexanderBabel** [comment]
The difference is that the video doorbell can show an unlock button if the lock and the camera are in the same room.

**Comment by krmkrx** [comment]
They are in the same room, no unlock button available to me.
Should I set the value for doorbell to false and only videoDoorbell to true?

**Comment by lukasroegner** [comment]
Hi Alex,

first of all, thank you for your contribution! I cannot test this version at the moment as haven't installed my nello yet (moved to another apartment). I have a question concerning the homekitUser: is it necessary to use a different user? I registered the plugin at nello.io so that the commands are sent via clientId authentication instead of a user action. This means if the plugin performs an unlock, it is not made behalf of a user. That's why there should be no doubled notifications.

If you think the pull request is ready for production, please tell me. Also, if you would like to be a contributor instead of forking and making pull requests, just contact me!

Best regards
Lukas

**Comment by AlexanderBabel** [comment]
It makes no difference. I catch it in the code.

I will look up the issue in the coming days an keep you updated.

**Comment by krmkrx** [comment]
I can provide you with more details once I get back home and resume testing.
Thank you for time time @us09alex !

**Comment by AlexanderBabel** [comment]
Hey Lukas,

currently I'm experiencing many issues with local tunnel. After a few hours my Homebridge instance crashes because of it. Therefore I will write a backend handling the web hook request. (This project will be open source). 

Duplicated notifications: I will take a look if it is possible to detect a difference between the unlocking action performed via the Home App and a action performed via the Nello app.

I think by the end of this weekend the code is ready for production.

Greetings
Alex

**Comment by lukasroegner** [comment]
Yea, a did some research concerning the web hook issue when I implemented that, local tunnel seemed to be the most simple solution that is free to use.

I'm looking forward to your implementation. Again, thank you very much for your work!

**Comment by krmkrx** [comment]
Problem resolved, only doorbell or videoDoorbell can be true.

**Comment by AlexanderBabel** [comment]
Hey @krmkrx,

that's great.

I pushed another update to the fork. Now the webhook handling is no longer via localtunel. Instead I used my own backend (https://github.com/us09alex/nello-backend)

Feel free to try it and to report bugs or errors.

Thanks in advance.

Greetings
Alex

**Comment by AlexanderBabel** [comment]
Hey @lukasroegner,

Unfortunately Nello don't send any information over the web hook if the unlock happened over the API or not. The body sent over the web hook is for both actions the same.

Therefore the best possibility to prevent doubled notifications is to use a separated user for this plugin.

Also I fixed some bugs from the last version, restructured your code and the new backend is up and running.

I also tested the plugin today and yesterday without any issue. The Homebridge instance didn't crashed or something else like before with local tunnel.

So I think this fork is ready for production.

Also I would help you to maintain this repo as contributor as you suggested earlier.

Greetings
Alex

**Comment by lukasroegner** [comment]
Hi Alex,

I understand the issue. Maybe we find another solution some time later, for now, I like the idea of using a separate account.

Could you add my latest changes to the pull request (if no configuration is provided, just don't initialize)?

I think you can also approve the PR as you are now a collaborator of the repo.

**Comment by AlexanderBabel** [comment]
Hey Lukas,

thanks for the invitation.

The change from you will be kept during merge.

Thanks for your trust but, I think its better when take a look at the changes before merging. Especially take a look at the readme file.

Greetings
Alex

**Comment by krmkrx** [comment]
Thank you for your work @us09alex @lukasroegner .
Version without localtunel working great.

Quick question about the username for the seperate accounting running this plugin.
Since the nello usernames consist of first name and last name, in which order do they need to be put into the config? And what would be the difference between the already available name string and the new homekitUser string? I am running a dedicated nello user for this plugin already. Therefore I assume that name and homekitUser need the same value?

**Comment by AlexanderBabel** [comment]
Hey there,

the order is "[first name][space][last name]".

The name property is for the name you see in the log files.
The homekitUser is the name of your dedicated account.

Greetings
Alex

**Comment by lukasroegner** [comment]
Hi Alex,

I went through the code and it looks fine to me. One think we should consider: Maybe we should add a note to the README that the relay service (your implementation for the webhook issue) is running on your server (e.g. at the "security" section), just to make sure that people are informed about it.

**Comment by AlexanderBabel** [comment]
Okay, 

I'll add this.

Can you give me the permission to publish the plugin at npmjs.org?

My username is there the same as here.

Greetings
Alex

**Comment by AlexanderBabel** [comment]
Closes #8

**Pull request merged**